scripts/utils/pre-release.mjs

```diff
@@ -0,0 +1,370 @@
+import path from 'path';
+import { createRequire } from 'module';
+import { fileURLToPath } from 'url';
+import moment from 'moment';
+import replace from 'replace-in-file';
+import inquirer from 'inquirer';
+import semver from 'semver';
+import {
+  displayErrorMessage,
+  displayConfirmationMessage
+} from './index.mjs';
+import JSDOMGlobal from 'jsdom-global';
+import chalk from 'chalk';
+import glob from 'glob';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+// Using `require` exclusively to import `json` files. Prevents 'experimental' warnings from being thrown in the
+// console.
+const require = createRequire(import.meta.url);
+const hotPackageJson = require('../../package.json');
+const workspacePackages = hotPackageJson.workspaces.packages;
+
+/**
+ * Check if the provided version number is a valid semver version number.
+ *
+ * @param {string} version Version number.
+ * @returns {boolean} `true` if the version number is a valid semver version number, `false` otherwise.
+ */
+export function isVersionValid(version) {
+  return !!semver.valid(version);
+}
+
+/**
+ * Check if the provided release date in a format of 'DD/MM/YYYY' is a valid future date.
+ *
+ * @param {string} date The date in format of 'DD/MM/YYYY'.
+ * @returns {object} Object containing information about the release validity in a form of `{valid: boolean, error:
+ * string}`.
+ */
+export function validateReleaseDate(date) {
+  const dateObj = moment(date, 'DD/MM/YYYY');
```

Comment:
```suggestion
  const dateObj = moment(date, 'DD/MM/YYYY', true);
```
The strict format has to be enabled. Otherwise, the validation returns true on dates such as `10/10.2030`.